src/script.js

```diff
@@ -111,22 +107,28 @@ class TodoView extends Croquet.View {
   addTodo() {
     const title = document.getElementById("newTodo").value;
     if (!title) return;
-
-    // Clear the input field
     newTodo.value = "";
 
-    // Optimistic local update
-    this.redraw({ todoId: Infinity, title, checked: false });
-
-    // Publish event to the model, and by extension, all views, including ours
     this.publish("todo", "add", { title });
   }
 
-  todoCheckButtonClicked(event) {
+  addedTodo({ todoId, title }) {
+    this.appendTodoItem({ todoId, title, checked: false });
+  }
+
+  toggleTodoCompletion(event) {
```

Comment:
@codefrau you mentioned in abb38654571316eea8883bc6ddf4c4d74ecc76c9 that we should try not to use the same name between the model and the view. Why? Is there a technical reason, or just stylistic?

Reply:
See my extended comment on this topic [here](https://github.com/jessmartin/croquet-todo-example/pull/4#issuecomment-934735123)

Reply:
No technical reason.

I personally find it easier to follow the code if entities that do different things are named differently. E.g. I would stick to "item" to refer to the model entity and "element" in the view to the DOM element. Same goes for method names - adding a todo item is different from adding a todo element. The model operates on different data structures than the view, etc.

Secondly, my IDE of choice allows to hover on an identifier and say "peek at definition". With distinctly named methods it will show the right one without further interaction.

Reply:
> I would stick to "item" to refer to the model entity and "element" in the view to the DOM element.

Good point. I'll rename.

> adding a todo item is different from adding a todo element

I do like the distinction. Hammer home the difference between model and view and what they typically operate on.

> Secondly, my IDE of choice allows to hover on an identifier and say "peek at definition". With distinctly named methods it will show the right one without further interaction.

Very reasonable and pragmatic. 😄